js/src/index.ts

```diff
@@ -6,36 +6,51 @@
 //
 //
 
-import { 
-    Connection, 
+import {
+    Connection,
     Keypair,
-    PublicKey, 
+    PublicKey
 } from '@solana/web3.js';
-import { 
-    NautilusQuery, 
+import { NautilusIdl, NautilusProgram } from './types';
+import {
+    NautilusQuery,
     NautilusTable,
 } from './sql';
-import { NautilusUtils } from './util';
 
-export class Nautilus {
+import { decapitalizeFirstLetter } from './util';
+
+export class Nautilus<Program extends NautilusProgram = NautilusProgram> {

```

Comment:
since a `Nautilus` instance is extending `NautilusProgram`, that would mean we are binding `Nautilus` instances 1-to-1 with programs, right?
This implementation is good and makes sense, but I can't shake the idea of - in the future - one `Nautilus` instance being able to operate on multiple programs, including non-Nautilus programs that have an IDL (Anchor, Shank)